extensions/amp-form/0.1/test/test-amp-form.js

```diff
@@ -330,7 +331,9 @@ describes.repeated('', {
         emailInput.setAttribute('required', '');
         form.appendChild(emailInput);
         sandbox.spy(form, 'checkValidity');
-        sandbox.stub(ampForm.xhr_, 'fetch').returns(Promise.resolve());
+        const {promise, stub} = stubWithCalledPromise(
```

Comment:
That looks good, but wouldn't polling would reintroduce flakiness? This implementation definitively resolves the promise and allows the test to continue as soon as the stubbed method is called.